### ISSUE TITLE
Allow clone in addition to unzip

### ISSUE THREAD

**Issue by xogeny** [xogeny/impact#53]
It would be good if `impact` optionally could clone in addition to unzipping dependencies.  I'm not sure how we should indicate when to clone and when to unzip, so we should think about that.

**Comment by xogeny** [comment]
[This looks promising](https://github.com/gogits/git), a pure Go implementation of git (read-only) functionality.

We already have the repository information in the index, BTW.  But I'm not sure under what circumstances we should actually pull from there?  Normally, we have a specific version number in mind.  How can we know that a) cloning from a repository will give us something compatible with the version we would otherwise checkout and b) that the user wants to checkout from the VC system vs a normal release.  These are actually the questions that are holding up implementation.  Comments?

**Comment by dietmarw** [comment]
I agree and that by enabling checkout as source we kind of opening up a whole can of worms. We should probably discuss with people interested in the feature how they would expect it to work. Pushing to post-conference for now.
